lib/clients/websocket.js

```diff
@@ -17,13 +17,11 @@ class WebsocketClient extends EventEmitter {
     { channels = null } = {}
   ) {
     super();
-    this.productIDs = Utils.determineProductIDs(productIDs);
+    this.productIDs = new Set(Utils.determineProductIDs(productIDs));
     this.websocketURI = websocketURI;
     this.auth = Utils.checkAuth(auth);
-    this.channels = channels || ['full'];
-    if (!this.channels.includes('heartbeat')) {
-      this.channels.push('heartbeat');
-    }
+    this.channels = new Set(channels || ['full']);
```

Comment:
`channels` right now can also be objects with `name` and `product_id` properties, this breaks this.